core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -2272,8 +2272,14 @@ && sameTypeOrNarrowsNullability(e.getType(), intExpr.getType())) {
           (RexLiteral) e.getOperands().get(1))
           : rexBuilder.makeCast(e.getType(), operand, safe, safe);
       executor.reduce(rexBuilder, ImmutableList.of(simplifiedExpr), reducedValues);
-      return requireNonNull(
-          Iterables.getOnlyElement(reducedValues));
+      RexNode reducedRexNode = requireNonNull(Iterables.getOnlyElement(reducedValues));
+      if (reducedRexNode instanceof RexCall) {
```

Comment:
it looks to me dangerous to not check the call type here. It looks like you assume it's always a Cast.
Maybe at least assert that.

Reply:
+1 
add CAST judgement `RexNode.isA(SqlKind.CAST)`

Reply:
@mihaibudiu, @YiwenWu 
Thanks for the review, I changed this to check whether the reduced expression is as CAST.

Reply:
@kasakrisz You should not squash directly. You need to wait until all reviewers have reviewed before squashing. Otherwise, the reviewer will not know what you changed.